src/platform/packages/shared/kbn-esql-utils/src/utils/append_to_query.ts

```diff
@@ -153,3 +153,13 @@ export const appendStatsByToQuery = (queryString: string, column: string) => {
     return `${queryString}\n| STATS BY ${column}`;
   }
 };
+
+export const appendLimitToQuery = (queryString: string, limit: number) => {
```

Comment:
Let's change it to this

```
export const appendLimitToQuery = (queryString: string, limit: number) => {
  const { root } = Parser.parse(queryString);
  mutate.commands.limit.upsert(root, limit);
  return BasicPrettyPrinter.print(root);
};
```

We have an api already for this. The BasicPrettyPrinter.print(root) is also removing the comments but you dont need this for your case